code/game/objects/items/weapons/cosmetics.dm

```diff
@@ -491,3 +491,71 @@
 		return
 	if (prob(25))
 		shatter()
+
+/obj/item/weapon/nanitecontacts
+	name = "Nanite Contacts"
+	desc = "Deploys nanobots to your eyes to change their color."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "nanite_contact"
+	flags = FPRINT
+	w_class = W_CLASS_TINY
+	var/color_r = 255
+	var/color_g = 255
+	var/color_b = 255
+
+/obj/item/weapon/nanitecontacts/New()
+	..()
+	color_r = rand(0,255)
+	color_g = rand(0,255)
+	color_b = rand(0,255)
+	update_icon()
+
+/obj/item/weapon/nanitecontacts/update_icon()
+	overlays.len = 0
+	var/icon/contact_color = new/icon("icon" = 'icons/obj/items.dmi', "icon_state" = "contacts_overlay")
+	contact_color.Blend(rgb(color_r, color_g, color_b), ICON_ADD)
+	overlays += contact_color
+
+/obj/item/weapon/nanitecontacts/attack_self(mob/user as mob)
+	var/new_color = input(user, "Choose the contact's color:", "Color Select") as color|null
+	if(new_color)
+		color_r = hex2num(copytext(new_color, 2, 4))
+		color_g = hex2num(copytext(new_color, 4, 6))
+		color_b = hex2num(copytext(new_color, 6, 8))
+	update_icon()
+
+/obj/item/weapon/nanitecontacts/attack(mob/M as mob, mob/user as mob)
+	if(!istype(M, /mob))
+		return
+
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		var/area = user.zone_sel.selecting
+		var/area_string = "eyes"
```

Comment:
Why is this necessary? Is this ever going to be applied to anything but eyes? I guess just replace occurrences of this with "eyes"